js/laserraster.js

```diff
@@ -244,13 +244,15 @@ Rasterizer.prototype.rasterRow = function(y) {
 
         // Convert Pixel Position to millimeter position
         posx = (posx * this.config.spotSize1 + parseFloat(this.config.xOffset));
-        posx = posx.toFixed(1);
+        posx = posx.toFixed(3);
         // Keep some stats of how many pixels we've processed
         this.megaPixel++;
 
         // The Luma grayscale of the pixel
-        var lumaGray = (pixels[x*4]*0.3 + pixels[x*4+1]*0.59 + pixels[x*4+2]*0.11)/255.0;
-        this.grayLevel = lumaGray.toFixed(1);
+        var lumaGray;
+	if (pixels[x*4+3] == 0) lumaGray = 1.0; // If full transparency => 0 intensity
```

Comment:
What if the alpha channel is 0.5? It would be better to do proper compositing OVER white background, like lumaGray = alpha * lumaGray + (1.0-alpha) * 1.0;